src/domain/ledger/index.types.d.ts

```diff
@@ -211,42 +227,56 @@ interface ILedgerService {
 
   getTransactionById(
     id: LedgerTransactionId,
-  ): Promise<LedgerTransaction | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency> | LedgerServiceError>
 
   getTransactionsByHash(
     paymentHash: PaymentHash | OnChainTxHash,
-  ): Promise<LedgerTransaction[] | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency>[] | LedgerServiceError>
 
   getTransactionsByWalletId(
     walletId: WalletId,
-  ): Promise<LedgerTransaction[] | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency>[] | LedgerServiceError>
 
   getTransactionsByWalletIdAndContactUsername(
     walletId: WalletId,
     contactUsername: Username,
-  ): Promise<LedgerTransaction[] | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency>[] | LedgerServiceError>
 
   listPendingPayments(
     walletId: WalletId,
-  ): Promise<LedgerTransaction[] | LedgerServiceError>
+  ): Promise<LedgerTransaction<WalletCurrency>[] | LedgerServiceError>
 
   listAllPaymentHashes(): AsyncGenerator<PaymentHash | LedgerError>
 
   getPendingPaymentsCount(walletId: WalletId): Promise<number | LedgerServiceError>
 
   getWalletBalance(walletId: WalletId): Promise<CurrencyBaseAmount | LedgerServiceError>
 
-  allPaymentVolumeSince(args: IGetVolumeArgs): VolumeResult
+  getWalletBalanceAmount<S extends WalletCurrency>({
+    walletId,
+    walletCurrency,
+  }: {
```

Comment:
this should be type `WalletDescriptor`

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1237/commits/511b884c10e209fa657008b92e18af3b4a4cf368